### PR TITLE
Feature/checkboxes component

### PR DESCRIPTION
**Pull request by njseeto**
This PR adds the checkbox component and validation for when the user does not provide an answer.

There is a work around for the validation as the method `govuk_collection_check_boxes` does not allow us to pass `include_hidden: false`, which means if a user does not answer an array with an empty string is passed thereby not triggering the validations. 

An issue has been raised for this and we will need to update this feature once the issue has been solved: https://github.com/DFE-Digital/govuk_design_system_formbuilder/issues/238

Story: https://trello.com/c/eOQZs8J2/1238-add-checkboxes-component-to-a-single-question-page

### CODE REVIEW

**Comment by brenetic** [comment]
i think we're gonna have to go with the plural `checkboxes` i'm afraid. :(

there are two schemas that need to be migrated

checkboxes: https://github.com/ministryofjustice/fb-components/blob/master/specifications/checkboxes/checkboxes.schema.json

and checkbox: https://github.com/ministryofjustice/fb-components/blob/master/specifications/checkbox/checkbox.schema.json

the checkboxes schema references the checkbox item on line 16. Currently it's referencing the radio item schema.

we'll also need default metadata for both checkboxes and a checkbox item.  the PR for the radios component hopefully illustrates what i mean: https://github.com/ministryofjustice/fb-metadata-presenter/pull/58/files

File: schemas/definition/checkbox.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    "label": {
+      "skipOptional": true
+    },
+    "hint": {
+      "title": "Hint text",
+      "description": "Text to help users understand an option - appears in grey under the label",
+      "type": "string",
+      "content": true
+    },
+    "conditionalComponent": {
```
Comment (brenetic) [inline schemas/definition/checkbox.json:23]: `conditional_component`

File: schemas/definition/checkbox.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      "$ref": "definition.name"
+    }
+  ],
+  "required": [
+    "name",
+    "label"
+  ],
+  "transforms": {
+    "namespace": {
+      "propagation": "conditionalComponent"
```
Comment (brenetic) [inline schemas/definition/checkbox.json:44]: `conditional_component`

File: schemas/component/checkboxes.json [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{
+  "$id": "http://gov.uk/schema/v1.0.0/checkboxes",
+  "_name": "checkboxes",
```
Comment (brenetic) [inline schemas/component/checkboxes.json:3]:
```suggestion
  "_name": "component.checkboxes",
```

File: schemas/definition/checkbox.json [hunk 4]
```diff
@@ -0,0 +1,47 @@
+{
+  "$id": "http://gov.uk/schema/v1.0.0/checkbox",
+  "_name": "checkbox",
```
Comment (brenetic) [inline schemas/definition/checkbox.json:3]:
```suggestion
  "_name": "definition.checkbox",
```

File: default_metadata/component/checkbox.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "_id": "component.checkbox",
+  "_type": "checkbox",
+  "errors": {},
+  "hint": "Component hint",
+  "items": [],
+  "name": "component-name",
+  "legend": "Required legend"
+}
```
Comment (brenetic) [inline default_metadata/component/checkbox.json:9]: so this file will need to be `checkboxes.json`

and in that case we're missing a `default_metadata/definition/checkbox.json`

Comment (brenetic) [reply default_metadata/component/checkbox.json:9]: also you're default metadata for the checkboxes should add two items by default. similar to the radios.json

https://github.com/ministryofjustice/fb-metadata-presenter/blob/main/default_metadata/component/radios.json

File: default_metadata/component/checkboxes.json [hunk 6]
```diff
@@ -0,0 +1,9 @@
+{
+  "_id": "definition.checkboxes",
+  "_type": "checkbox",
```
Comment (brenetic) [inline default_metadata/component/checkboxes.json:3]:
```suggestion
  "_type": "checkboxes",
```

sorry! didn't spot that the last time

File: default_metadata/component/checkboxes.json [hunk 7]
```diff
@@ -0,0 +1,9 @@
...
+  "errors": {},
+  "hint": "Component hint",
```
Comment (brenetic) [inline default_metadata/component/checkboxes.json:5]:
```suggestion
  "hint": "[Hint text (optional)]",
```

double check the prototype, but i think that's the text fabien wanted

File: default_metadata/component/checkboxes.json [hunk 8]
```diff
@@ -0,0 +1,9 @@
...
+  "items": [],
+  "name": "component-name",
+  "legend": "Required legend"
```
Comment (brenetic) [inline default_metadata/component/checkboxes.json:8]:
```suggestion
  "legend": "Question"
```

again, i think that's what's in the prototype

File: fixtures/version.json [hunk 9]
```diff
@@ -223,6 +223,49 @@
...
       ]
     },
+    {
+      "_uuid": "7b748584-100e-4d81-a54a-5049190136cc",
+      "_id": "page.burgers",
+      "_type": "page.singlequestion",
+      "components": [
+        {
+          "_id": "burgers_checkboxes_1",
+          "_type": "checkbox",
```
Comment (brenetic) [inline fixtures/version.json:233]:
```suggestion
          "_type": "checkboxes",
```

File: default_metadata/definition/checkbox.json [hunk 10]
```diff
@@ -0,0 +1,7 @@
+{
+	"_id": "definition.checkbox",
+	"_type": "checkbox",
+	"label": "Option",
+	"hint": "[Hint text (optional)]",
+	"value": "checkbox-value"
+}
```
Comment (tomas-stefano) [inline default_metadata/definition/checkbox.json:7]: @brenetic can you confirm if this definition file should be on the schema?

Comment (tomas-stefano) [reply default_metadata/definition/checkbox.json:7]: Oh wait, I think this file is not needed.

Comment (brenetic) [reply default_metadata/definition/checkbox.json:7]: for collections we would have:

- schema for the collection
- schema for the collection item
- default metadata for the collection
- default metadata for the collection item

File: default_metadata/component/checkboxes.json [hunk 11]
```diff
@@ -0,0 +1,9 @@
+{
+  "_id": "definition.checkboxes",
```
Comment (brenetic) [inline default_metadata/component/checkboxes.json:2]:
```suggestion
  "_id": "component.checkboxes",
```

File: default_metadata/definition/checkbox.json (hunk 10 above)
Comment (tomas-stefano) [reply default_metadata/definition/checkbox.json:7]: Oh I see. True. Then we can access the attribute of a new item via `DefaultMetadata['definition.checkbox']`
I think the word definition just confused me. Apologies.

Comment (tomas-stefano) [reply default_metadata/definition/checkbox.json:7]: Thanks for the clarification

Comment (brenetic) [reply default_metadata/definition/checkbox.json:7]: we do need the default metadata for a collection item as we will be inserting it into a button

**Review by brenetic — APPROVED** [review]
🚀

**Pull request merged**